Datum/Public/Resolve-DatumPath.ps1

```diff
@@ -30,28 +30,38 @@ function Resolve-DatumPath {
             $StackItem = [scriptblock]::Create( ($StackItem -f ([string[]]$PathVariables)) ).Invoke()
             Write-Debug -Message ($StackItem | Format-List * | Out-String)
             $PathItem = $stackItem
-        }
-        else {
+            $property = $CurrentNode.psobject.properties.where( {$_.name -eq $StackItem.Keys})
+        } else {
             $PathItem = $CurrentNode.($ExecutionContext.InvokeCommand.ExpandString($StackItem))
+            $property = $CurrentNode.psobject.properties.where( {$_.name -eq $StackItem})
+        }
+
+        if ($property -and $property.value -isnot [DatumProvider] -and $CurrentNode -is [DatumProvider]) {
+            $getter = $Property.GetterScript.ToString()
+            $PathItem = ([scriptblock]::Create(("$getter -AsJsonString")).Invoke())
+            if ([string]::IsNullOrWhiteSpace($PathItem)) {
+                $PathItem = $null
+            } else {
+                $PathItem = $PathItem | ConvertFrom-Json | ConvertTo-Datum
```

Comment:
There's quite a bit of a performance cost to call ConvertTo-Datum unfortunately...